utils/monitoring.js

```diff
@@ -0,0 +1,113 @@
+const { errors } = require('arsenal');
+const promClient = require('prom-client');
+const { Registry } = require('prom-client');
+const CountMaster = require('../CountItems/CountMaster');
+
+const aggregatorRegistry = new promClient.AggregatorRegistry();
+const { collectDefaultMetrics } = promClient;
+
+const bucketCount = (promClient.register.getSingleMetric('bucketCount_metrics'))
+    || new promClient.Gauge({
+        name: 'bucketCount_metrics',
+        help: 'Total number of buckets processed',
+        labelNames: ['bucketName'],
+    });
+
+// Histogram of the bucket processing duration, by the utilization service.
+
+const bucketProcessingDuration = (promClient.register.getSingleMetric('count_items_bucketProcessingDuration'))
+    || new promClient.Histogram({
+        name: 'count_items_bucketProcessingDuration',
+        help: 'Bucket processing duration',
+        labelNames: ['service'],
+        buckets: [1, 10, 60, 600, 3600, 18000, 36000],
+    });
+
+const consolidationDuration = (promClient.register.getSingleMetric('count_items_consolidationDuration'))
+    || new promClient.Histogram({
+        name: 'count_items_consolidationDuration',
+        help: 'Duration of metrics consolidation in seconds',
+        labelNames: ['service'],
+        buckets: [0.01, 0.05, 0.1, 0.2, 0.5, 1, 1.5, 2],
+    });
+
+const workersCount = (promClient.register.getSingleMetric('count_items_workersCountErrors'))
+    || new promClient.Counter({
+        name: 'count_items_workersCountErrors',
```

Comment:
```suggestion
        name: 's3_countitems_worker_errors_count',
```

...though it does not seem to be an error in "worker" : the metric should not represent "where in the code in fails", but "what operation" did actually fail...
→ a "worker" error would be for exemple if we failed to create a worker, or if a worker crashed
→ in this case, this is used when `getObjectMDStats` returns an error, i.e. when we fail to process a bucket: so it may be more a case of `s3_countitems_total_buckets_count` with a label indicating success or failure?

Reply:
renamed here : https://github.com/scality/s3utils/pull/312/commits/a50c483216bfb00c5fa7bc9b25c02e1cbb673bbb#diff-612a4c10e405510cdaad879589d4a7d157df6e0ff42b4645d2a21d4d17e41fa9R24-R28